### PR TITLE
Fix ci

### PR DESCRIPTION
**Pull request by Shourya742**
* Fix Clippy warning
* Fix docker setup

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -50,11 +50,11 @@ jobs:
         run: cargo fmt -- --check
 
   docker_build:
-    name: Docker Build
+    name: Docker Compose Build
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
```
Comment (GitGab19) [inline .github/workflows/ci.yaml:57]: Why did you remove `macos`?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -63,24 +63,18 @@ jobs:
...
-      - name: Build Docker Images
-        run: |
-          docker build -t log-server ./log-server
-          docker build -t pools-latency-calculator ./pools-latency-calculator
-          docker build -t sv1-custom-proxy ./sv1-custom-proxy
-          docker build -t sv2-custom-proxy ./sv2-custom-proxy
-          docker build -f sv1-public-pool.dockerfile -t sv1-public-pool .
-          docker build -f template-provider.dockerfile -t template-provider .
-          docker build -f sv2-roles.dockerfile -t sv2-roles .
-          docker build -f pools-latency-calculator.dockerfile -t pools-latency-calculator .
```
Comment (GitGab19) [inline .github/workflows/ci.yaml:86]: Why did you remove them?
How are they built now?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -63,24 +63,24 @@ jobs:
...
+          sudo mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --batch --yes --dearmor -o /etc/apt/keyrings/docker.gpg
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+            $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+          sudo apt-get update
+          sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+          sudo curl -L "https://github.com/docker/compose/releases/download/v2.20.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+          sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
```
Comment (GitGab19) [inline .github/workflows/ci.yaml:80]: Why not just purely following the official guide here: https://docs.docker.com/engine/install/ubuntu/

Comment (Shourya742) [reply .github/workflows/ci.yaml:80]: I had this available from one of my previous CI files, so I just took it from there.

File: log-server/src/main.rs [hunk 4]
```diff
@@ -21,12 +22,14 @@ struct Data {
     result: Vec<ResultItem>,
 }
 
+#[allow(dead_code)]
```
Comment (GitGab19) [inline log-server/src/main.rs:25]: Why did you introduce this if you tackled the clippy warning we were getting?

File: log-server/src/main.rs [hunk 5]
```diff
@@ -21,12 +22,14 @@ struct Data {
...
 #[derive(Deserialize, Debug)]
 struct ResultItem {
     stream: Stream,
     values: Vec<(String, String)>,
 }
 
+#[allow(dead_code)]
```
Comment (GitGab19) [inline log-server/src/main.rs:32]: Same here

File: log-server/src/main.rs [hunk 6]
```diff
@@ -182,8 +185,10 @@ async fn fetch_logs(
 
     let formatted_logs: String = logs
         .into_iter()
-        .map(|(_, message)| format!("{}\n", message))
-        .collect();
+        .fold(String::new(), |mut acc, (_, message)| {
+            writeln!(&mut acc, "{}", message).unwrap();
+            acc
+        });
```
Comment (GitGab19) [inline log-server/src/main.rs:191]: Is this change needed?

File: sv2-custom-proxy/src/main.rs [hunk 7]
```diff
@@ -296,7 +297,11 @@ async fn connect_to_server(server_address: &str) -> TcpStream {
 }
 
 pub fn encode_hex(bytes: &[u8]) -> String {
-    bytes.iter().map(|b| format!("{:02x}", b)).collect()
+    let hex_string = bytes.iter().fold(String::new(), |mut acc, b| {
+        write!(&mut acc, "{:02x}", b).unwrap();
+        acc
+    });
+    hex_string
```
Comment (GitGab19) [inline sv2-custom-proxy/src/main.rs:304]: Same here

File: log-server/src/main.rs (hunk 4 above)
Comment (Shourya742) [reply log-server/src/main.rs:25]: Because we do require this to parse the payload but we never actually use it. So, having it as a dead underling is somewhat justified.

File: log-server/src/main.rs (hunk 5 above)
Comment (Shourya742) [reply log-server/src/main.rs:32]: same as above

File: sv2-custom-proxy/src/main.rs (hunk 7 above)
Comment (Shourya742) [reply sv2-custom-proxy/src/main.rs:304]: same as above

File: log-server/src/main.rs (hunk 4 above)
Comment (GitGab19) [reply log-server/src/main.rs:25]: Can you point me where do you parse the payload?
To me it seems completely unused

File: sv1-custom-proxy/src/main.rs [hunk 8]
```diff
@@ -182,6 +181,7 @@ async fn transfer(
     Ok(())
 }
 
+#[allow(unused_assignments)]
```
Comment (GitGab19) [inline sv1-custom-proxy/src/main.rs:184]: Which was the unused one? Shouldn't we eventually clean it up?

**Review by GitGab19 — APPROVED** [review]
